integration_test/ops_agent_test.go

```diff
@@ -4140,6 +4163,29 @@ func TestBufferLimitSizeOpsAgent(t *testing.T) {
 	})
 }
 
+// TestNoNvmlOtelReceiverWithoutGpu check to make sure that Ops Agent's
+// hostmetrics receiver does not generate a nvml otel receiver on VMs without
+// GPU
+func TestNoNvmlOtelReceiverWithoutGpu(t *testing.T) {
+	t.Parallel()
+	gce.RunForEachPlatform(t, func(t *testing.T, platform string) {
+		t.Parallel()
+
+		ctx, logger, vm := agents.CommonSetup(t, platform)
+		if err := agents.SetupOpsAgent(ctx, logger.ToMainLog(), vm, ""); err != nil {
+			t.Fatal(err)
+		}
+
+		if otelConf, err := getGeneratedOtelConfig(ctx, logger, vm); err != nil {
```

Comment:
Hm, this feels pretty fragile to me (baking in knowledge of the generated otel configs). Maybe we can just check to make sure there is no mention of `nvml` in Cloud Logging?

See e.g. `TestCustomLogFile`, which tests for the absence of a log like this:

```go
                time.Sleep(60 * time.Second)
                _, err := gce.QueryLog(ctx, logger.ToMainLog(), vm, "mylog_source", time.Hour, `jsonPayload.message="abc test pattern xyz"`, 5)
                if err == nil {
                        t.Error("expected log to be excluded but was included")
                } else if !strings.Contains(err.Error(), "not found, exhausted retries") {
                        t.Fatalf("unexpected error: %v", err)
                }
```

Reply:
Done!